avogadro/qtplugins/svg/svg.cpp

```diff
@@ -0,0 +1,259 @@
+}
+
+// get all model, view, image positions sort them and save id to vector index
+void SVG::getPositions()
+{
+  calculateCamera();
+  m_mols.clear();
+  const Core::Array<Vector3> mols = m_molecule->atomPositions3d();
+  Eigen::Vector3f dummy = m_camera->modelView() * mols[0].cast<float>();
+  dummy[1] *= -1.0f;
+  m_min = dummy;
+  m_max = dummy;
+
+  for (unsigned int i = 0; i < mols.size(); ++i) {
+    Eigen::Vector3f pos = m_camera->modelView() * mols[i].cast<float>();
+    pos[1] *= -1.0f;
+    Mol mol(mols[i].cast<float>(), pos, i, m_molecule->bonds(i).size());
+    m_mols.push_back(mol);
+    if (frustrumCulling(mol)) {
+      for (unsigned int j = 0; j < 3; ++j) {
+        if (m_min[j] > pos[j])
+          m_min[j] = pos[j];
+        if (m_max[j] < pos[j])
+          m_max[j] = pos[j];
+      }
+    }
+  }
+  m_min[0] -= IMAGE_PADDING;
+  m_min[1] -= IMAGE_PADDING;
+  m_max[0] += IMAGE_PADDING;
+  m_max[1] += IMAGE_PADDING;
+  m_max -= m_min;
+  std::sort(m_mols.begin(), m_mols.end(), [&](const Mol& a, const Mol& b) {
+    return a.pos_view.norm() > b.pos_view.norm();
+  });
+  int i = 0;
+  for (auto& m : m_mols) {
+    m_idToindex[m.id] = i;
+    m.pos_image = posToImage(m.pos_view, m.r);
+    ++i;
+  }
+}
+
+// paint all bondings only once
+void SVG::paintBonds(QPainter& painter, const Mol& mol, unsigned int i,
+                     const NeighborListType& bonds)
+{
+  painter.setPen(QPen(QColor(125, 125, 125), PEN_WIDTH_BOND));
+  for (auto it = bonds.begin(); it != bonds.end(); ++it) {
+    unsigned int j = it->atom2().index();
+    if (m_idToindex[j] == i) {
+      j = it->atom1().index();
+    }
+    if (m_idToindex[j] <= i) {
+      continue;
+    }
+    // calculate the inicial and final position, considering the mol middle
+    // and the middle of the pen width
+    auto mol_to = m_mols[m_idToindex[j]].pos_image;
+    Eigen::Vector2f from(mol.pos_image[0] - (PEN_WIDTH_BOND / 2.0f),
+                         mol.pos_image[1] - (PEN_WIDTH_BOND / 2.0f));
+    Eigen::Vector2f to(mol_to[0] - (PEN_WIDTH_BOND / 2.0f),
+                       mol_to[1] - (PEN_WIDTH_BOND / 2.0f));
+
+    float L = std::sqrt((from[0] - to[0]) * (from[0] - to[0]) +
+                        (from[1] - to[1]) * (from[1] - to[1]));
+    float offsetX = (to[1] - from[1]) / L;
+    float offsetY = (from[0] - to[0]) / L;
+    unsigned int order = int(it->order());
+    // for each bound offset it following the orthogonal direction
+    for (unsigned int o = 0; o < order; ++o) {
+      // if there is only one bond, don't displace
+      float x = 0, y = 0;
+      if (order > 1) {
+        x = (float(o) - (order / 2.0f)) * OFF_SET_PARALEL * offsetX;
+        y = (float(o) - (order / 2.0f)) * OFF_SET_PARALEL * offsetY;
+      }
+      QLineF line(from[0] + x, from[1] + y, to[0] + x, to[1] + y);
+      painter.drawLine(line);
+    }
+  }
+}
+
+void SVG::paintMol(QPainter& painter, const Mol& mol)
+{
+  Vector3ub color_mol = m_molecule->color(mol.id);
```

Comment:
See, "mol" is usually taken as a short word for "molecule", but this seems to be:
`Vector3ub atom_color = m_molecule->color(atom.id)`